packages/flutter/lib/src/widgets/tap_and_drag_gestures.dart

```diff
@@ -979,6 +960,7 @@ class TapAndDragGestureRecognizer extends OneSequenceGestureRecognizer with _Tap
             // but the pointer has exceeded the tap tolerance, then the pointer is accepted as a
             // drag gesture.
             if (currentDown != null) {
+              _dragState = _DragState.accepted;
```

Comment:
I am wondering why `_pastSlopTolerance && _wonArenaForPrimaryPointer` should accept drag?
But why `_pastSlopTolerance && !_wonArenaForPrimaryPointer` need reject the gesture instead of waiting the gesture  win result to accetp drag?

Reply:
In other words, what should we do if `_pastSlopTolerance == true` and has not won the arena (will won later)?

Reply:
My logic here is because `_pastSlopTolerance == true` the gesture cannot be considered a tap. And because we are in `didStopTrackingLastPointer` for the given `pointer` and we have still not passed the drag threshold then the gesture cannot be considered a drag so we cancel and reject.

`DragGestureRecognizer` has this logic as well.
https://github.com/flutter/flutter/blob/a28aae918eaefb266482c1a3b33955d2f26ed119/packages/flutter/lib/src/gestures/monodrag.dart#L458-L460.

Reply:
Got it.